backends/platform/android/events.cpp

```diff
@@ -1106,43 +1106,59 @@ void OSystem_Android::pushEvent(int type, int arg1, int arg2, int arg3,
 	case JE_GAMEPAD:
 		switch (arg1) {
 		case AKEY_EVENT_ACTION_DOWN:
-			e.type = Common::EVENT_KEYDOWN;
+			e.type = Common::EVENT_JOYBUTTON_DOWN;
```

Comment:
What effect could it have on the existing engines?

Reply:
I don't have the knowledge to speak to specific engines, but I can say that previously only one button could be used in the button mapper (X specifically, since it was sending an ESC keycode).

I might be mistaken, but since the joypad is mapped to virtual mouse movement and A and B joypad buttons are mapped to left and right click by default, the behavior in-game hasn't really changed for those despite switching the event types to ones that can be re-mapped. I think the only way an engine could break is if it relied 100% on the physical mouse input which the buttons were previously emulating, and didn't have a way to map other inputs to those mouse events.